core/base-service/route.js

```diff
@@ -44,23 +44,29 @@ function prepareRoute({ base, pattern, format, capture, withPng }) {
   return { regex, captureNames }
 }
 
-function namedParamsForMatch(captureNames = [], match, ServiceClass) {
-  // Assume the last match is the format, and drop match[0], which is the
-  // entire match.
-  const captures = match.slice(1, -1)
-
-  if (captureNames.length !== captures.length) {
+function paramsForReq(captureNames = [], req, ServiceClass) {
```

Comment:
- Rewrite for express, where `req.params` is an object like `{1: 'foo', 2: 'bar'}`.
- Unpack `namedParams` and `format` in one place, which feels like better encapsulation (and is DRYer too).